src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -259,7 +262,10 @@ def verify_index_connection(self) -> None:
             ) from e
 
     def create_canopy_index(self,
-                            indexed_fields: Optional[List[str]] = None,
+                            spec: Union[Dict, ServerlessSpec, PodSpec] = ServerlessSpec(
+                                cloud='aws',
+                                region='us-west-2'
```

Comment:
We should probably make these arguments of `create_caopy_index()` (with strong defaults.

Reply:
Oh, sorry, I was missing the point that this is part of the function definition...

In this case, the pythonic way would be `sepc: [Optional[Union[Dict, ServerlessSpec, PodSpec]] = None`, and then inside `__init__()` define set this default spec you crated here.